src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -95,20 +137,40 @@ public PackageArchiveReader(ZipArchive zipArchive)
         /// <param name="zipArchive">ZipArchive containing the nupkg data.</param>
         /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing.</param>
         /// <param name="compatibilityProvider">Framework compatibility provider.</param>
-        public PackageArchiveReader(ZipArchive zipArchive, IFrameworkNameProvider frameworkProvider, IFrameworkCompatibilityProvider compatibilityProvider)
+        /// <param name="requiredRepoSign">Set to true if the package is needed to be repository signed.</param>
+        public PackageArchiveReader(
+            ZipArchive zipArchive,
+            IFrameworkNameProvider frameworkProvider,
+            IFrameworkCompatibilityProvider compatibilityProvider,
+            bool packageSignatureVerified = false,
+            bool requiredRepoSign = false,
+            IEnumerable<IRepositoryCertInfo> repositoryCertInfos = null)
             : base(frameworkProvider, compatibilityProvider)
         {
             _zipArchive = zipArchive ?? throw new ArgumentNullException(nameof(zipArchive));
+            PackageSignatureVerified = packageSignatureVerified;
+            RequiredRepoSign = requiredRepoSign;
+            RepositoryCertInfos = repositoryCertInfos;
         }
 
-        public PackageArchiveReader(string filePath, IFrameworkNameProvider frameworkProvider = null, IFrameworkCompatibilityProvider compatibilityProvider = null)
+        public PackageArchiveReader(
+            string filePath,
+            IFrameworkNameProvider frameworkProvider = null,
+            IFrameworkCompatibilityProvider compatibilityProvider = null,
+            bool packageSignatureVerified = false,
+            bool requiredRepoSign = false,
+            IEnumerable<IRepositoryCertInfo> repositoryCertInfos = null)
             : base(frameworkProvider ?? DefaultFrameworkNameProvider.Instance, compatibilityProvider ?? DefaultCompatibilityProvider.Instance)
         {
             if (filePath == null)
             {
                 throw new ArgumentNullException(nameof(filePath));
             }
 
+            PackageSignatureVerified = packageSignatureVerified;
+            RequiredRepoSign = requiredRepoSign;
+            RepositoryCertInfos = repositoryCertInfos;
```

Comment:
where and how are you using these fields from package reader? shouldn't these only be with extractor?